crates/recursor/src/recursor.rs

```diff
@@ -88,6 +90,20 @@ impl RecursorBuilder {
         self
     }
 
+    /// Add recommended IPs to the do not query list
+    pub fn recommended_filters(mut self, recommended_filters: bool) -> Self {
```

Comment:
So it looks like you have the code defaulting to including the recommended filters at the config level, but not at the `RecursorBuilder` level, where you have to actively opt-in?

For consistency, shouldn't we just make this opt-out instead of opt-in, that is, should the default be to enable these filters and then allow users to override the filters (and perhaps we can expose the list of recommended filters so users can use them to filter down/augment our recommendations)?

Reply:
I tried to simplify this in the most recent commit (which I'll keep separate for now and squash it in if the approach ends up working out.  A couple of big changes:

There are only two configuration knobs now: query_allow and query_block (formerly do_not_query).

All of the recommended ranges are added to query_block automatically.  Anything the user wants to allow as an exception can be added to query_allow (i.e., if an IP matches query_allow and query_block, query_allow wins.)

So, if a user is comfortable with our recommended filters but wants to block an additional range (say 1.1.1.0/24, they configure:

```
query_block = ["1.1.1.0/24"]
```

If they want to create a granular exception to our defaults, (allowing 192.168.1.0/24 but leaving the rest of 192.168.0.0/16 blocked,) they configure:

```
query_allow = ["192.168.1.0/24"]
```

If they want to completely disable our filters:

```
query_allow = ["0.0.0.0/0", "::/0"]
```

If they want to block a new range but allow a narrow exception:

```
query_allow = ["1.1.1.1/32"]
query_block = ["1.1.1.0/24"]
```

I *think* this will be easier to deal with in practice than trying to expose the default filter list and allow exceptions, which seemed like it would require something like

```
exclude_from_default_filters = "192.168.0.0/16"
```

or

```
[[filter_exceptions]]
rfc_1918_192_168_0_0_16 = false
```

The first option I think would be error-prone in practice; the second would be really verbose.